frame/offences/src/lib.rs

```diff
@@ -80,11 +90,22 @@ decl_event!(
 );
 
 decl_module! {
-	/// Offences module, currently just responsible for taking offence reports.
 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
 		fn deposit_event() = default;
+
+		fn on_initialize(_now: T::BlockNumber) {
+			// only decode storage if we can actually submit anything again.
+			if T::OnOffenceHandler::can_report() {
+				// defensive only. Per definition, `OnOffenceHandler` should accept reports when
```

Comment:
I'd remove this comment, the assumption is pretty weak, we don't have a ways to enforce that if `can_report` returns `false` the actual reporting won't fail.